lib/Runtime/Library/SameValueComparer.h

```diff
@@ -31,39 +31,56 @@ namespace Js
             }
         }
 
+        static hash_t HashDouble(double d)
+        {
+            if (JavascriptNumber::IsNan(d))
+            {
+                return 0;
+            }
+
+            if (zero)
+            {
+                // SameValueZero treats -0 and +0 the same, so normalize to get same hash code
+                if (JavascriptNumber::IsNegZero(d))
+                {
+                    d = 0.0;
+                }
+            }
+
+            __int64 v = *(__int64*)&d;
```

Comment:
I don't believe so.  Only NaN has multiple representations.  Number values are all uniquely represented, including the denormalized numbers.  Since we canonicalize NaN we don't have to worry about accidentally giving NaN multiple hash values.

Actually, I wonder then, we probably can also get rid of that special case for NaN above.